crates/data_components/src/graphql/provider.rs

```diff
@@ -69,7 +69,11 @@ impl GraphQLTableProviderBuilder {
         self
     }
 
-    pub async fn build(self, query_string: &str) -> Result<GraphQLTableProvider> {
+    pub async fn build(
+        self,
+        query_string: &str,
+        inspection_query: Option<GraphQLQuery>,
```

Comment:
Add this to the builder itself, i.e. `GraphQLTableProviderBuilder::new(..).with_inspection_query().build()`.